### PR TITLE
Refactor function names

### PR DESCRIPTION
**Pull request by ixfoduap**
Function names in PL-Qchem were previously long and repetitive. This PR changes function names to shorten then and make them more intuitive. Note that file names for tests have also been changed

### CODE REVIEW

File: qchem/pennylane_qchem/qchem.py [hunk 1]
```diff
@@ -531,34 +531,34 @@ def generate_mol_hamiltonian(
...
 
-    docc_indices, active_indices = gen_active_space(
+    docc_indices, active_indices = active_space(
         mol_name, hf_data, n_active_electrons, n_active_orbitals
     )
 
     return (
-        gen_hamiltonian_pauli_basis(mol_name, hf_data, mapping, docc_indices, active_indices),
+        decompose_hamiltonian(mol_name, hf_data, mapping, docc_indices, active_indices),
         2 * len(active_indices),
```
Comment (ixfoduap) [inline qchem/pennylane_qchem/qchem.py]: Is this comma needed after `2 * len(active_indices)` ?

Comment (josh146) [reply qchem/pennylane_qchem/qchem.py]: I think it's optional? is the returned tuple of size 3?

File: qchem/pennylane_qchem/qchem.py [hunk 2]
```diff
@@ -531,34 +531,34 @@ def generate_mol_hamiltonian(
...
     "read_structure",
-    "gen_meanfield_data",
-    "gen_active_space",
-    "gen_hamiltonian_pauli_basis",
+    "meanfield_data",
+    "active_space",
+    "decompose_hamiltonian",
     "_qubit_operator_to_terms",
     "_terms_to_qubit_operator",
     "_qubit_operators_equivalent",
```
Comment (josh146) [inline qchem/pennylane_qchem/qchem.py]: @ixfoduap, I vaguely remembered you saying some of these private functions might be useful for the user?

File: qchem/pennylane_qchem/qchem.py (hunk 1 above)
Comment (agran2018) [reply qchem/pennylane_qchem/qchem.py]: No

File: qchem/pennylane_qchem/qchem.py (hunk 2 above)
Comment (agran2018) [reply qchem/pennylane_qchem/qchem.py]: These private functions were implemented by Zy to convert convert OpenFermion ``QubitOperator`` to PennyLane ``Hamiltonian`` operator and viceversa and for testing the function ``convert_hamiltonian``.

File: qchem/pennylane_qchem/qchem.py (hunk 1 above)
Comment (ixfoduap) [inline qchem/pennylane_qchem/qchem.py]:
```suggestion
        2 * len(active_indices)
```

Comment (ixfoduap) [reply qchem/pennylane_qchem/qchem.py]: Somehow black adds the comma, so I guess I'll keep it

File: qchem/pennylane_qchem/qchem.py (hunk 2 above)
Comment (josh146) [reply qchem/pennylane_qchem/qchem.py]: :+1:

**Pull request merged**